### DISCUSSION TITLE
Having problem with events...

### DISCUSSION THREAD

**Discussion by mikend** [admob-plus/admob-plus#348] [Q&A] [answered]
I'm having problems listening to events in ionic/vue...

This stops execution with no error message:

    interstitial.on('dismiss', async () => {
        console.log('dismissed')
    })

Also tried this but no success:

    document.addEventListener('admob.interstitial.dismiss', async () => {
        console.log('dismissed')
    })

What am I doing wrong?

Thanks

**Comment by mikend** [comment, marked as answer]
Ok, I seem to have solved it. :-)

**Reply by ratson** [reply]
@mikend How did you solve it? It would be helpful for others for others to learn from.

**Comment by lithiumhax** [comment]
I'm also having problems with the Events.  Any update would help.

**Comment by Taz-Bizeloper** [comment]
Hi,

I am also having issues with the same event. It fires on the first time you show the Interstitial Ad. Firing the Ad the second time no longer calls dismiss event or any event. Java throws an error "Application attempted to call on a destroyed WebView" 

Any help would be appreciated.

I am not sure if I am doing something wrong the way I create/show the Ad. Below is my code. I have tried to make the interstitial varaible instatiated one time and I also tried to re-instantiate it every time but still no luck.

    this.ShowInterstitial = function () {

            interstitial = new admob.InterstitialAd({
                adUnitId: 'ca-app-pub-3940256099942544/8691691433',// TEST UNIT
            });
            interstitial.on('load', (evt) => {
                // evt.ad
            });
            interstitial.on('impression', (evt) => {
                // evt.ad
                console.log(evt);
            });
            document.addEventListener('admob.interstitial.dismiss', function () {
                  console.log("DISMISSED ");
            });

        return interstitial.load().then(function () {
            return interstitial.show().then(function (res) {
                return res;
            });
        });
    }
